tests/unit/s2n_fork_generation_number_test.c

```diff
@@ -189,6 +190,9 @@ static int s2n_unit_test_clone_child_process(void *parent_process_fgn)
     /* Verify in threads */
     EXPECT_EQUAL(s2n_unit_test_thread(return_fork_generation_number), S2N_SUCCESS);
 
+    /* Clean up urandom of the child process */
+    POSIX_GUARD_RESULT(s2n_rand_cleanup());
```

Comment:
As mentioned in https://github.com/aws/s2n-tls/pull/4831#discussion_r1793994571, should this just be calling s2n_cleanup?

Reply:
We discussed this offline - s2n_cleanup() doesn't work here since atexit is enabled, so s2n_cleanup() won't actually invoke the rand cleanup callback to close urandom.

Reply:
Refer to this [comment](https://github.com/aws/s2n-tls/pull/4835#discussion_r1811132231) for further details.